src/common/Socket.cpp

```diff
@@ -0,0 +1,40 @@
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include "Socket.h"
+#include "Utils.h"
+
+Socket::Socket(int socket_fd)
+    : m_socket_fd(socket_fd)
+{
+}
+
+Socket::~Socket()
+{
+    ::close(m_socket_fd);
```

Comment:
dtor code should always be wrapped in try catch(...)
the exception that thrown inside a dtor can cause unknown shit to the calling stack